### ISSUE TITLE
[Question] Connection between group_id and "meta" column type

### ISSUE THREAD

**Issue by Sharan123** [reczoo/FuxiCTR#104]
First of all thank you for the hard work and the very nice library you are all developing! 

I am trying to better understand the _group_id field inside the model_config and the "meta" column_ type which we set in the dataset_config. I would like to extend the library to support field losses and fields in general (like losses and models in "[Field-aware Calibration: A Simple and Empirically Strong Method for Reliable Probabilistic Predictions](https://arxiv.org/abs/1905.10713)" ) 

So basically we need to be able to pass the categorical values of a certain column (ideally set in model_config) in the model.forward (and model.evaluate_metrics) function like so
```python
def forward(self, inputs):
    # code
    return_dict = {
         "y_pred": y_pred,
         "logit": logit, 
         "field": col_values
         }
    return return_dict
```

**How would I be able to best achieve this such to follow the library construction?** This way we can use it to train the model while accounting for samples belonging to different categories. This looks very similar to group_id field in the code (except this would be related to field and it would be used to control the model loss function, as well as implementing new field-related metrics). 

In reading the code I came upon the group_id and the related "meta" column type but not much examples on how they should be used in the library.  

Related to this: 
1. Does the group_id in the model_config need to be connected to a "meta" column? (or can it be any type of column, e.g. "categorical")
2. Is meta column used for training? If it is, what is the difference between the categorical and meta variables as both of them have an underlying tokenizer, and in general how are they handled differently? 
3. Since I would like to split samples based on a categorical column, I assume that I would need to have the same column as both meta and categorical column (categorical for training and meta for splitting - similar as group id in the group metrics) ? In which case are we able to use the "meta" column for loss function and metric evaluation but discard it from the training features?

In general any clarification on how I could expand the library to include the new features and loss while following your design would be much appreciated!

**Comment by zhujiem** [comment]
> How would I be able to best achieve this such to follow the library construction? 

It is simple. We get X, a dict of input columns. and thus you get pick a column from it.
```
X = self.get_inputs(inputs)
    return_dict = {
         "y_pred": y_pred,
         "logit": logit, 
         "field": X["col_name"]
         }
```

For `meta` type, we define it as the input type that is not used as features as input, but instead as a type of side information for statisitcs. So `group_id` is only used for computing gAUC during evaluation, but it is not fed into the model forward.

**Comment by zhujiem** [comment]
To rewrite the loss function, please refer to https://github.com/reczoo/FuxiCTR/blob/main/model_zoo/FinalNet/src/FinalNet.py#L106
